mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -1375,13 +1364,13 @@ impl MakerSwap {
         Ok((swap, command))
     }
 
-    pub async fn recover_funds(&self) -> Result<RecoveredSwap, String> {
-        async fn try_spend_taker_payment(selfi: &MakerSwap, secret_hash: &[u8]) -> Result<TransactionEnum, String> {
+    pub async fn recover_funds(&self) -> MmResult<RecoveredSwap, RecoverSwapError> {
```

Comment:
There is a place which calls `recover_funds` taker and maker methods wrapped in `try_s` macro 

https://github.com/KomodoPlatform/komodo-defi-framework/blob/10e41928c3f394ef88f4ffddcb5a99ad1dde256c/mm2src/mm2_main/src/lp_swap/saved_swap.rs#L111-L120

as `try_s!` appends error path + MmErr also contains error path, Im afraid there will be unreadable result error message with error path duplications
```rust
Ok(try_s!(maker_swap.recover_funds().await))
Ok(try_s!(taker_swap.recover_funds().await))
```

I would suggest to extract MmErr inner messages and put it into `ERR!` in `SavedSwap.recover_funds()`. 
Or return `MmResult` in `recover_funds` from `SavedSwap` and in RPC method `fn recover_funds_of_swap`. `recover_funds` from `SavedSwap` is used inly in `recover_funds_of_swap`

Reply:
ummmm, i imagined we would have both paths as in kind of nesting, which is good (and was the behaviour before this PR anyways).
i will try to trigger this and see if it looks complicated then i'll simplify it.

Reply:
upd:  its non blocking and not urgent note. I suppose we have other places which already have same issue. I think it will be fixed in refactored kdf repo.